lib/Subscriber/Phpcr/RemoveSubscriber.php

```diff
@@ -55,12 +56,20 @@ public static function getSubscribedEvents()
      * remove any references to the node.
      *
      * @param RemoveEvent $event
+     *
+     * @throws DocumentReferencedException
      */
     public function handleRemove(RemoveEvent $event)
     {
         $document = $event->getDocument();
         $node = $this->documentRegistry->getNodeForDocument($document);
 
+        $references = iterator_to_array($node->getReferences());
```

Comment:
Just did a quick search, the only thing I could do is using `iterator_count` here, don't know how much of a difference this is in terms of performance.